extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -0,0 +1,164 @@
+  renderShare_() {
+    const urlParams = {};
+
+    for (const param in this.typeConfig_['params']) {
+      const paramConf = this.typeConfig_['params'][param];
+      let paramValue = this.config_[param] || this.getDefaultValue_(
+          param, this.config_);
+      AMP.assert(!paramConf['required'] || paramValue !== null,
+        param + ' is a required attribute for ' + this.type_ + '. %s',
+        this.element);
+      if (paramValue == null && paramConf['type'] == 'fixed') {
+        paramValue = paramConf['value'];
+      }
+      if ('maxlength' in paramConf) {
+        const maxlength = paramConf['maxlength'];
+        AMP.assert(!paramValue || paramValue.length < maxlength,
+          param + ' cannot exceed ' + maxlength + '. %s', this.element);
+      }
+      if (paramValue != null) {
+        urlParams[paramConf['param']] = paramValue;
+      }
+    }
+
+    // Get the anchor or create one.
+    let link = elementByTag(this.element, LINK_TAG);
+    if (link === null) {
+      link = this.getWin().document.createElement(LINK_TAG);
+      link./*OK*/innerHTML = this.typeConfig_['text'];
+      link.setAttribute('target', '_blank');
+
+      // Get the container or create one.
+      let container = elementByTag(this.element, CONTAINER_TAG);
+      if (container === null) {
+        container = this.getWin().document.createElement(CONTAINER_TAG);
+        container.classList.add('-amp-social-share');
+        container.classList.add('-amp-social-share-' + this.type_);
+
+        // Only add the container to the element if it didn't exist
+        this.element.appendChild(container);
+      }
+      container.appendChild(link);
+    }
+
+    // Set share url.
+    link.setAttribute('href',
+      addParamsToUrl(this.typeConfig_['url'], urlParams));
+  }
+
+  /**
+   * Gets the configuration for the current social element
+   * @param {!element} element
+   * @return {?Object}
+   */
+  getElementConfig_() {
+    const script = elementByTag(this.element, 'script');
+    let config = {};
+    if (script) {
+      // Get config from script
+      try {
+        config = JSON.parse(script.textContent);
+      } catch (e) {
+        AMP.assert(false, 'Malformed JSON configuration. %s', this.element);
+      }
+    } else {
+      // Get config from attributes
+      for (const param in this.typeConfig_['params']) {
+        if (this.element.hasAttribute('data-' + param)) {
+          config[param] = this.element.getAttribute('data-' + param);
+        }
+      }
+    }
+    return config;
+  }
+
+  /**
+   * Gets the default value for a given param, if there is one.
+   * Otherwise it returns null
+   * @param  {!string} param
+   * @param  {!Object} config
+   * @return {*}
+   */
+  getDefaultValue_(param, config) {
+    if (param in config) {
+      return config[param];
+    }
+    switch (param) {
+      case 'url':
+        return getSourceUrl(window.location.href);
+      case 'text':
+        return window.document.title;
```

Comment:
Just `''` may also be a good default. Do you like it if the text is in the post? It typically is redundant with the embed.